packages/runtime/src/schema.ts

```diff
@@ -1311,11 +1326,18 @@ class SchemaBuilder {
 
     this.#mutations[`update${name}`] = mutationWithClientMutationId({
       name: `Update${name}`,
-      inputFields: () => ({
-        id: { type: new GraphQLNonNull(GraphQLID) },
-        content: { type: new GraphQLNonNull(this.#inputObjects[`Partial${name}`]) },
-        options: { type: UpdateOptionsInput },
-      }),
+      inputFields: () => {
+        const inputFields: GraphQLInputFieldConfigMap = {
+          id: { type: new GraphQLNonNull(GraphQLID) },
+          options: { type: UpdateOptionsInput },
+        }
+        // It's possible content can't be updated if all fields are immutable
```

Comment:
Shouldnt we throw an error if an update mutation on an object with only immutable fields is attempted?

Reply:
With this change the mutation input wouldn't have any `content` object so no field could be changed, but as we'll add the `shouldIndex` property to the `options` it should still be possible to toggle indexing.